doc/interoperability/pandas.rst

```diff
@@ -0,0 +1,299 @@
+>>> X, y = iris.data[:-1,:], iris.target[:-1]
+>>> iris_pd = pd.DataFrame(X)
+>>> iris_pd.columns = iris.feature_names
+>>> iris_pd['target'] = y
+>>>
+>>> from sklearn.model_selection import train_test_split
+>>> train, test = train_test_split(iris_pd, test_size= 0.3, random_state=42)
+>>>
+>>> feature_columns_train = ['sepal length (cm)','sepal width (cm)',
+...                          'petal length (cm)','petal width (cm)']
+>>> # last two correct order
+>>> feature_columns_test = ['sepal length (cm)','sepal width (cm)',
+...                         'petal width (cm)','petal length (cm)']
+>>> # last two switched order
+>>>
+>>> from sklearn.linear_model import LogisticRegression
+>>> lg = LogisticRegression(n_jobs=4, random_state=123, verbose=0,
+...                         penalty='l2', C=1.0,
+...                         solver='lbfgs', multi_class='auto')
+>>> lg.fit(train[feature_columns_train], train['target'])
+LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
+                   intercept_scaling=1, l1_ratio=None, max_iter=100,
+                   multi_class='auto', n_jobs=4, penalty='l2', random_state=123,
+                   solver='lbfgs', tol=0.0001, verbose=0, warm_start=False)
+>>>
+>>> prob1 = lg.predict_proba(test[feature_columns_train])
+>>> prob1[:5]
+array([[4.11775575e-03, 8.20651654e-01, 1.75230591e-01],
+       [9.42197387e-01, 5.78021174e-02, 4.96014488e-07],
+       [2.80792378e-07, 5.36112664e-03, 9.94638593e-01],
+       [6.92583081e-03, 7.79167290e-01, 2.13906879e-01],
+       [1.64170075e-03, 7.43560544e-01, 2.54797756e-01]])
+>>> # result is actually
+>>> prob2 = lg.predict_proba(test[feature_columns_test])
+>>> prob2[:5]
+array([[7.97326265e-01, 1.74068589e-01, 2.86051460e-02],
+       [9.96001477e-01, 3.99849230e-03, 3.08998201e-08],
+       [2.50514227e-03, 7.47234078e-03, 9.90022517e-01],
+       [7.15353463e-01, 2.33443544e-01, 5.12029929e-02],
+       [5.67215000e-01, 3.41931708e-01, 9.08532926e-02]])
+
+
+At the time of writing, it is the users responsibility to ensure that the
+column ordering in the data used for training the estimator is the same as the
+ordering of the data used for prediction. There is an ongoing discussion
+whether or not this will change in the future and this
+`issue <https://github.com/scikit-learn/scikit-learn/issues/7242>`__ should be
+watched and used to update this paragraph in the future. A simple and straight-
+forward way of ensuring that column ordering and column labels are the same is
+using something like :meth:`df.loc[:, list of column names]` to enforce the
+correct ordering.
+
+Handling Categorical data
+=========================
+
+Section to be extended.
+
+See the following references to get started:
+
+- https://scikit-learn.org/stable/glossary.html#term-categorical-feature
+- https://scikit-learn.org/stable/modules/preprocessing.html#preprocessing-categorical-features
+- https://github.com/scikit-learn-contrib/sklearn-pandas
```

Comment:
It would be nice to expand this section, now that OneHotEncoder/OrdinalEncoder support non-numeric data (but very short pointing to the user guide is good)

Can you use sphinx syntax to link to the other section in the docs? (see http://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations)

If you have time of course!

Reply:
I have addressed all of your other comments. I will work on this section over the next days and ping you when it is updated. Have a look at the rest and see if they are ok now.